coordinator/src/collaborative_revert.rs

```diff
@@ -25,257 +26,280 @@ use dlc_manager::subchannel::LnDlcChannelSigner;
 
-    let sub_channel = sub_channels
+    let subchannel = subchannels
         .iter()
         .find(|c| c.channel_id == channel_id)
-        .context("Could not find provided channel")?;
+        .context("Missing subchannel")?;
 
-    let position =
-        Position::get_position_by_trader(&mut conn, channel_details.counterparty.node_id, vec![])?
-            .context("Could not load position for channel_id")?;
+    let peer_id = subchannel.counter_party;
+    let fund_txo_sat = subchannel.fund_value_satoshis;
+
+    // The app's unspendable reserve is almost certainly 1% of the funding TXO, although this
+    // depends on the `their_channel_reserve_proportional_millionths` config value set by the
+    // coordinator for this channel.
+    let unspendable_punishment_reserve_sat = match channel_details_option {
+        Some(ref details) => details.counterparty.unspendable_punishment_reserve,
+        None => {
+            let fund_txo_sat = Decimal::from(fund_txo_sat);
 
-    let settlement_amount = position
-        .calculate_settlement_amount(revert_params.price)
-        .context("Could not calculate settlement amount")?;
-
-    let trade_settled = sub_channel.fund_value_satoshis == channel_details.channel_value_satoshis;
-
-    // There is no easy way to get the total tx fee for all subchannel transactions, hence, we
-    // estimate it. This transaction fee is shared among both users fairly
-    let dlc_channel_fee = calculate_dlc_channel_tx_fees(
-        trade_settled,
-        sub_channel.fund_value_satoshis,
-        channel_details.inbound_capacity_msat / 1000,
-        channel_details.outbound_capacity_msat / 1000,
-        position.trader_margin as u64,
-        position.coordinator_margin as u64,
-        channel_details.unspendable_punishment_reserve.unwrap_or(0),
-    )?;
-
-    // Coordinator's amount is the total channel's value (fund_value_satoshis) whatever the taker
-    // had (inbound_capacity), the taker's PnL (settlement_amount) and the transaction fee
-    let coordinator_amount = match trade_settled {
-        false => {
-            sub_channel.fund_value_satoshis as i64
-                - (channel_details.inbound_capacity_msat / 1000) as i64
-                - settlement_amount as i64
-                - (dlc_channel_fee as f64 / 2.0) as i64
-                - channel_details.unspendable_punishment_reserve.unwrap_or(0) as i64
+            let reserve = fund_txo_sat / Decimal::ONE_HUNDRED;
+
+            // The minimum reserve is _at least_ 1_000 sats.
+            let reserve = reserve.max(Decimal::ONE_THOUSAND);
+
+            reserve.to_u64().expect("Reserve to fit into u64")
         }
-        true => {
-            sub_channel.fund_value_satoshis as i64
-                - (channel_details.inbound_capacity_msat / 1000) as i64
-                - channel_details.unspendable_punishment_reserve.unwrap_or(0) as i64
+    };
+
+    let is_channel_split = match channel_details_option {
```

Comment:
🤔 Is this always correct? Comparing the balances might not be as elegant, but it was simply and worked.

Reply:
Yeah, this should work the same way and is simpler. The `funding_outpoint` and the channel balances are updated at the same time when calling `set_funding_outpoint`.